### PR TITLE
pdns: fix usage of notify only when zone kind is Master or Slave 

### LINKED ISSUE #1780: pdns: Regression with changes introduced in #1679
### Welcome

- [X] Yes, I'm using a binary release within 2 latest releases.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).

### What did you expect to see?

A dns challenge for a pdns native zone doesn't not work anymore, whereas it worked in previous version.

### What did you see instead?

Changes introduced in #1679 only works when pdns Zone are kind of Master or Slave.

Updating a pdns native zone doesn't works anymore, because of the notify request where server respond HTTP 400 as [documentation explains]()https://doc.powerdns.com/authoritative/http-api/zone.html#put--servers-server_id-zones-zone_id-notify.

### How do you use lego?

Through Traefik

### Reproduction steps

1. Create a native zone une pdns
2. Try to and let's encrypt certifactes with dns challenge on this zone
3. Wait for Traefik to show a error

### Version of lego

```console
4.9.0 and 4.9.1
```

### Logs

error: one or more domains had a problem:
```
[*.renamed.domain] [*.renamed.domain] acme: error presenting token: pdns: unexpected HTTP status code 400 when PUTing 'https://renamed-pdns-01.domain/api/v1/servers/localhost/zones/renamed.domain./notify'
[renamed.domain] [renamed.domain] acme: error presenting token: pdns: unexpected HTTP status code 400 when PUTing 'https://renamed-pdns-01.domain/api/v1/servers/localhost/zones/renamed.domain./notify'
```

### Go environment (if applicable)

_No response_

### PR DESCRIPTION
**Pull request by melck**
Hello,

Fixes #1780

It should fix linked issue. What do you think about it ?

### CODE REVIEW

File: providers/dns/pdns/pdns.go [hunk 1]
```diff
@@ -173,7 +173,7 @@ func (d *DNSProvider) Present(domain, token, keyAuth string) error {
 		return fmt.Errorf("pdns: %w", err)
 	}
 
-	if d.apiVersion < 1 {
+	if d.apiVersion < 1 || (zone.Kind != "Master" && zone.Kind != "Slave") {
```
Comment (dmke) [inline providers/dns/pdns/pdns.go:176]: Can you extract the logic into a helper function, and invert the logic here?

Something like

```go
if supportsZonetransfer(d.apiVersion, zone.Kind) {
   	err = d.notify(zone.URL)
	if err != nil {
		return fmt.Errorf("pdns: %w", err)
	}
}
return nil
```

File: providers/dns/pdns/pdns.go [hunk 2]
```diff
@@ -221,7 +221,7 @@ func (d *DNSProvider) CleanUp(domain, token, keyAuth string) error {
...
```
Comment (dmke) [inline providers/dns/pdns/pdns.go:224]: Same here.

**Review by dmke — CHANGES_REQUESTED** [review]
Looks good, but could be slightly improved.

The conditional got complex enough to be extracted into a helper. This allows us (or maybe you?) to add a simple test case for it.

**Review by ldez — APPROVED** [review]
LGTM

**Review by dmke — APPROVED** [review]

**Pull request merged**